File: Npc/11000353.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11000353: Thompson
/// </summary>
public class _11000353 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:0831180407001467$
    // - How may I help you?
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:0831180407001470$
                // - Our fuel needs are growing by the day. I think more excavation sites will be needed soon.
                return 30;
            case (30, 1):
                // $script:0831180407001471$
                // - We're pumping oil every day. It's still kind of surprising how much we need to get by.
                return -1;
        }
        
        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.Next,
            (30, 1) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11000353.py
""" 11000353: Thompson """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0831180407001467$
        # - How may I help you?
        return None # TODO

    def __30(self, pick: int) -> int:
        if self.index == 0:
            # $script:0831180407001470$
            # - Our fuel needs are growing by the day. I think more excavation sites will be needed soon.
            return 30
        elif self.index == 1:
            # $script:0831180407001471$
            # - We're pumping oil every day. It's still kind of surprising how much we need to get by.
            return -1
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.NEXT
        elif (self.state, self.index) == (30, 1):
            return Option.CLOSE
        return Option.NONE
